### ISSUE TITLE
time_t support update

### ISSUE THREAD

**Issue by genotrance** [genotrance/nimterop#37]
Right now, `time_t` is hard-coded to `int32`. Once devel becomes a release and perhaps n-1, use `time_t.Time` instead.

https://github.com/nim-lang/Nim/pull/10319

cc @zestyr

**Comment by zestyr** [comment]
This doesn't seem to be working for function definitions, it works for struct fields though.

```c
void foo(time_t time);

struct bar { 
    time_t time; 
};
```
```nim
type
  bar* {.importc: "struct bar", header: headertest, bycopy.} = object
    time*: int32

proc foo*(time: time_t) {.importc: "foo", header: headertest.}
```

**Comment by genotrance** [comment]
Fixed the issue in the latest commit. See https://github.com/genotrance/nimterop/issues/36 for some commentary.

Issue still needs to be open so that we can change time_t in the future.

**Comment by genotrance** [comment]
@timotheecour is improving this in PR #73 - this will allow 0.19.2 and devel to coexist but with a correct value for time_t.

We will fix this issue when we remove 0.19.2 code in types.nim once it is no longer supported.

**Comment by timotheecour** [comment]
> We will fix this issue when we remove 0.19.2 code in types.nim once it is no longer supported.

instead of keeping this issue open once https://github.com/genotrance/nimterop/pull/73 is merged, let's simply cleanup all code that uses `when (NimMajor, NimMinor, NimPatch) < (0, 19, 9)` once newer Nim is released
